lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2868,6 +2875,42 @@ ParseResult FIRStmtParser::parseStaticRefExp(Value &result,
                  parseOptionalExpPostscript(result, false));
 }
 
+/// path ::= 'path(' static_ref ')'
+// NOLINTNEXTLINE(misc-no-recursion)
+ParseResult FIRStmtParser::parsePath(Value &result) {
+  auto startTok = consumeToken(FIRToken::lp_path);
+
+  Value staticRef;
+  if (parseStaticRefExp(staticRef,
+                        "expected static reference expression in 'path'") ||
+      parseToken(FIRToken::r_paren, "expected ')' in 'path'"))
+    return failure();
+
+  locationProcessor.setLoc(startTok.getLoc());
+
+  auto fieldRef = getFieldRefFromValue(staticRef);
+  auto target = fieldRef.getValue();
+
+  // Ports are handled differently, emit a PathOp with inner symbol.
+  if (auto arg = dyn_cast<BlockArgument>(target)) {
```

Comment:
Create the target, then share the innerref + creation code? I have this locally for experimenting with using RWProbeOp exclusively instead of Forceable, FWIW.

Does this work for things like instances (probably not since we want a `Value` for `staticRef`)?

Reply:
It has been updated to work with instance ops, see the message down below.